pennylane/qchem/factorization.py

```diff
@@ -169,3 +170,145 @@ def factorize(two_electron, tol_factor=1.0e-5, tol_eigval=1.0e-5):
+        :math:`h_{pssq}`, as
+
+        .. math::
+
+            T_{pq} = h_{pq} - \frac{1}{2} \sum_s h_{pssq}.
+
+        The tensor :math:`V` can be converted to a matrix which is indexed by the indices :math:`pq`
+        and :math:`rs` and eigendecomposed up to a rank :math:`R` to give
+
+        .. math::
+
+            V_{pqrs} = \sum_r^R L_{pq}^{(r)} L_{rs}^{(r) T},
+
+        where :math:`L` denotes the eigenvectors of the matrix. The molecular Hamiltonian can then
+        be rewritten following Eq. (7) of
+        [`Phys. Rev. Research 3, 033055, 2021 <https://journals.aps.org/prresearch/abstract/10.1103/PhysRevResearch.3.033055>`_]
+        as
+
+        .. math::
+
+            H = \sum_{\alpha \in \{\uparrow, \downarrow \} } \sum_{pq} T_{pq} a_{p,\alpha}^{\dagger}
+            a_{q, \alpha} + \frac{1}{2} \sum_r^R \left ( \sum_{\alpha, \beta \in \{\uparrow, \downarrow \} } \sum_{pq}
+            L_{pq}^{(r)} a_{p, \alpha}^{\dagger} a_{q, \alpha} \right )^2.
+
+        The orbital basis can be rotated such that each :math:`T` and :math:`L^{(r)}` matrix is
+        diagonal. The Hamiltonian can then be written following Eq. (2) of
+        [`npj Quantum Information, 7, 23 (2021) <https://www.nature.com/articles/s41534-020-00341-7>`_]
+        as
+
+        .. math::
+
+            H = U_0 \left ( \sum_p d_p n_p \right ) U_0^{\dagger} + \sum_r^R U_r \left ( \sum_{pq}
+            d_{pq}^{(r)} n_p n_q \right ) U_r^{\dagger}
+
+        where the coefficients :math:`d` are obtained by diagonalizing the :math:`T` and
+        :math:`L^{(r)}` matrices. The number operators :math:`n_p = a_p^{\dagger} a_p` can be
+        converted to qubit operators using
+
+        .. math::
+
+            n_p = \frac{1-Z_p}{2}
+
+        where :math:`Z_p` is the Pauli :math:`Z` operator applied to qubit :math:`p`. This gives
+        the qubit Hamiltonian
+
+        .. math::
+
+           H = U_0 \left ( \sum_p O_p^{(0)} \right ) U_0^{\dagger} + \sum_r^R U_r \left ( \sum_{q} O_q^{(r)} \right ) U_r^{\dagger},
+
+        where :math:`O = \sum_i c_i P_i` is a linear combination of Pauli words :math:`P_` that are
+        a tensor product of Pauli :math:`Z` and Identity operators. This allows all the Pauli words
+        in each of the :math:`O` terms to be measured simultaneously. This function returns the
+        coefficients and the Pauli words grouped for each of the :math:`O` terms as well as the
+        eigenvectors of the :math:`T` and :math:`L^{(r)}` matrices that can be used to construct the
+        basis rotation unitaries :math:`U`.
+    """
+    two_electron = np.swapaxes(two_electron, 1, 3)
+
+    _, eigvals_m, eigvecs_m = qml.qchem.factorize(two_electron, tol_factor, 0.0)
+
+    t_matrix = one_electron - 0.5 * np.einsum("illj", two_electron)
+    t_eigvals, t_eigvecs = np.linalg.eigh(t_matrix)
+
+    eigvals = [np.array(t_eigvals)] + [np.outer(x, x).flatten() * 0.5 for x in eigvals_m]
+    eigvecs = [t_eigvecs] + eigvecs_m
```

Comment:
What doe this intend to do? `t_eigvecs` will be an array where its columns are the eigenvectors, but what about `eigvecs_m`?

Reply:
eigvecs is a list containing all the eigenvectors: those from the T matrix, which specify the one-body terms and those from factorising the two-body integrals.